app.py

```diff
@@ -66,13 +68,109 @@ def nodes_list():
     except Exception as e:
         return jsonify({"error": str(e)})
 
+def run_fulfillment_background(network_id, requested_nodes):
+    conn = get_esi_connection(cloud=cloud_name)
+    # Run async fulfillment tasks inside this thread
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    loop.run_until_complete(fulfill_order_loop(conn, network_id, requested_nodes))
+    loop.close()
+
+async def fulfill_order_loop(conn, network_id, requested_nodes):
+    # Keep in the loop until all nodes requests fulfilled
+    while True:
+        tasks = []
+        for item in requested_nodes:
+            resource_class = item['resource_class']
+            number = item['number']
+            if number <= 0:
+                continue
+            offers_all = conn.lease.offers(resource_class=resource_class)
+            # Check if the offer is available now
+            offers_available = [o for o in offers_all
+                                if check_offer_availability(o)]
```

Comment:
Is it possible to pass in `available_start_time` and `available_end_time` above and avoid this extra function?

Reply:
I tried to update it to o.available_start_time and o.available_end_time and it returned NoneType error. I think the offers.get_all() https://github.com/CCI-MOC/esi-leap/blob/fd5473dc4da6d55655388a824e9ec269af8e53c9/esi_leap/api/controllers/v1/offer.py#L205 doesn't have available_start_time and available_end_time in the return value.

Reply:
Oh, I meant in the API request - for example, https://github.com/CCI-MOC/python-esileapclient/blob/master/esileapclient/osc/v1/offer.py#L213 passes in `available_start_time` and `available_end_time` as filters

Reply:
Done! I've set the available_end_time to be 7 days from now. Is that acceptable?